evp.go

```diff
@@ -16,6 +16,17 @@ import (
 // cacheMD is a cache of crypto.Hash to GO_EVP_MD_PTR.
 var cacheMD sync.Map
 
+// hashFuncHash return fn(). If fn panics, it returns nil.
+// This is used to avoid aborting the program when calling
+// an unsupported hash function. It is the caller's responsibility
```

Comment:
Is the *only* possible panic due to lack of support? (It seems like this is the claim that would make this ok, and I think it deserves a direct mention and justification if possible.) Otherwise, preserving the error message seems like it would make sense because it would be important to figure other cases out--maybe even to know whether they're happening.

Reply:
Good point. I'll preserve the error wherever possible.